### ISSUE TITLE
Low test accuracy from pretraining_vgg_librispeech.py

### ISSUE THREAD

**Issue by QEDan** [ericguizzo/anti_transfer#1]
Thank you for your work on the paper "Blissful Ignorance: Anti-Transfer Learningfor Task Invariance" and for sharing your code on github. The approach is very interesting.

I am trying to work through the instructions in your code base, but I am having trouble reproducing your performance metrics with `pretraining_vgg_librispeech.py`. I have run the data preprocessing steps from the SpeechVGG repo as well as your `preprocessing_LIBRISPEECH.py` script. When I run `pretraining_vgg_librispeech.py`, the results are the following:

```
 Epoch: [8/50] [===================>] loss: 0.22886
  Train loss: 0.22193 | Val loss: 1.43724
  Train acc: 0.92656 | Val acc: 0.67556

 Epoch time: 3868.7 seconds
Training early-stopped

 train results:
train_loss: 0.22170693420957238
train_acc: 0.9267182026814833
train_f1: 0.8696645473487867
train_precision: 0.871358441153495
train_recall: 0.8708770562845035

 val results:
val_loss: 1.4339568866717238
val_acc: 0.6765427769985973
val_f1: 0.5253611039934988
val_precision: 0.5274069516997331
val_recall: 0.5329216266891146

 test results:
test_loss: 1.448861069614262
test_acc: 0.6628240694038132
test_f1: 0.5058537523372251
test_precision: 0.5073561778592891
test_recall: 0.5142333676621744
```

I also tried running again for all 50 epochs with early stopping deactivated, but it didn't not improve the val or test results substantially.

From Table 1 of your paper, I believe the `test_acc` value should be 91.85. The only modification that I made to the script was to set `pretraining_dataset_path` to the appropriate path on my system. I also modified `configANTITRANSFER.ini` to have `sequence_length=4` and `input_librispeech_folder` to the correct path on my system, as described in the `README` file. 

I hope that you can help me identify what I might be doing wrong. First, am I interpreting your tables correctly when I say that `test_acc` reported from this script should be 0.9185 from Table 1? Second, since the performance results that I achieve are low, is there any other modifications that I should make to the code or the configuration files to achieve the >90% accuracy using this script? Is there any mistake that I might have made with the preprocessing steps that could result in reduced performance?

Thank you so much for any help you can offer.

**Comment by ericguizzo** [comment]
Hello QEDan, thanks for your comment. 
This is very strange. Yes, you are correctly interpreting the tables, you should get ~97% train and ~92% test accuracy. 
- Are you using the 100 hours version of Librispeech?
- The values you are reporting are similar to the ones of the Nsynth pre-training, are you double sure that you are looking at the actual Librispeech results?

**Comment by ericguizzo** [comment]
Also, when the pre-training finishes, it saves a npy dict, reporting all actual results of the training. You should have it in pretraining_vgg/librispeech/4secs_inv/results.npy. 
Please, check if that file contains the expected results.

**Comment by QEDan** [comment]
Thanks for responding. The information and suggestions were helpful.

I have confirmed that I was looking at the LibriSpeech results and that the `pretraining_vgg/librispeech/4secs_inv/results.npy` file contained the same results as the terminal output.

I managed to find one problem. In my `LibriSpeech/split` directory, I had the `train-clean-100` subdirectory, but also the `train-clean-360` and `train-other-500` subdirectories. In that case, the `preprocessing_LIBRISPEECH.py` script loops over each of these and saves the resulting matrices to the same file name. In my case, it was one of the others that was persisted and the `train-clean-100` data had been overwritten. I fixed this by creating a copy of the data that only had the `train-clean-100` training subdirectory and not the other two.

When I use the proper training data, my results improve:
```
 Epoch: [14/50] [===================>] loss: 0.48798
  Train loss: 0.15045 | Val loss: 0.879
  Train acc: 0.94894 | Val acc: 0.805

 Epoch time: 3851.8 seconds
Training early-stopped

 train results:
train_loss: 0.15102538931946102
train_acc: 0.9486169827273824
train_f1: 0.9080680658443281
train_precision: 0.9094759743812139
train_recall: 0.9090261772092778

 val results:
val_loss: 0.8818351412263944
val_acc: 0.8044701986754967
val_f1: 0.6950407481141618
val_precision: 0.696816798028525
val_recall: 0.7002351893412478

 test results:
test_loss: 0.6376097123747883
test_acc: 0.8456045560747665
test_f1: 0.7404237272529849
test_precision: 0.7413224180404584
test_recall: 0.7456700866837391
```
Unfortunately, they only improve to test accuracy of 84.5%, which is still under-performing relative to 92%. 

My `LibriSpeech/split` directory looks like the following:
```
.../LibriSpeech/split$ ls
BOOKS.TXT     dev-clean    README.TXT    test-clean
CHAPTERS.TXT  LICENSE.TXT  SPEAKERS.TXT  train-clean-100
```
Is it correct to use `dev-clean` and `test-clean` instead of `dev-other` and `test-other`? 

The npy files that I get out of the pretraining then look like the following:
```
dataset/matrices/librispeech/librispeech$ ls -lh 
total 60G
2.6M     4sec_inv_training_target.npy
55G     4sec_inv_training_predictors.npy
134K     4sec_inv_test_target.npy
2.8G     4sec_inv_test_predictors.npy
142K    4sec_inv_validation_target.npy
3.0G     4sec_inv_validation_predictors.npy
```

Please let me know if any of the above looks incorrect. Thanks!
